File: 0053-Maximum Subarray/golang/solution.go
package golang

import "math"

func maxSubArray(nums []int) int {
	sum := 0
	solution := math.MinInt32
	for _, num := range nums {
		sum += num
		if num > sum {
			sum = num
		}
		if sum > solution {
			solution = sum
		}
	}

	return solution
}

// Runtime: 4 ms, faster than 97.88% of Go online submissions for Maximum Subarray.
// Memory Usage: 3.3 MB, less than 73.13% of Go online submissions for Maximum Subarray.

File: 0053-Maximum Subarray/c++/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0;
        int solution = INT_MIN;
        for(int num: nums) {
            sum = max(sum+num, num);
            solution = max(solution, sum);
        }
        return solution;
    }
};

// Runtime: 8 ms, faster than 97.34% of C++ online submissions for Maximum Subarray.
// Memory Usage: 13.2 MB, less than 68.04% of C++ online submissions for Maximum Subarray.
